src/hooks/useMasternodes.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from 'react'
+import { useWhaleApiClient } from '../layouts/contexts/WhaleContext'
+import { masternodes } from '@defichain/whale-api-client'
+
+interface useMasternodesData{
+  loading: boolean
+  masternodes: masternodes.MasternodeData[]
+  hasNext: boolean
+  next: string | number | undefined
+  error: boolean
+}
+
+export function useMasternodes (querySize: number, nextToken?: string): useMasternodesData {
+  const [loading, setLoading] = useState(true)
+  const [error, setError] = useState(false)
+  const [masternodes, setMasternodes] = useState<masternodes.MasternodeData[]>([])
+  const [hasNext, setHasNext] = useState<boolean>(false)
+  const [next, setNext] = useState<string | number | undefined>(undefined)
+
+  const api = useWhaleApiClient()
+
+  useEffect(() => {
+    setLoading(true)
+    setError(false)
+
+    // eslint-disable-next-line
+    api.masternodes.list(querySize, nextToken || undefined)
+      .then(res => {
+        setLoading(false)
+        setMasternodes(preMasternodes => [...preMasternodes, ...res])
+        setHasNext(res.hasNext)
+        setNext(res.nextToken)
+      })
+      .catch(e => {
+        setError(true)
+      })
+  }, [nextToken])
```

Comment:
```suggestion

  async function fetchMasternodes (size: number, next: string | undefined)  {
    try {
      const res = await api.masternodes.list(size, next)
      setLoading(false)
      setMasternodes(preMasternodes => [...preMasternodes, ...res])
      setHasNext(res.hasNext)
      setNext(res.nextToken)
    } catch {
      setError(true)
    }
  }

  useEffect(() => {
    setLoading(bah)
    setError(false)
    
    fetchMasternodes(querySize, nextToken)
  }, [nextToken])
```

Reply:
Actually we could also set the error message from the fetching masternodes into the `error` state.
We could provide some useful from the response to the user. If there's no need for that, we could still force it to be a boolean in the `masternodes` pages by checking for `!!error`